source_gen/lib/src/builder.dart

```diff
@@ -66,6 +66,12 @@ class _Builder extends Builder {
       throw ArgumentError(
           'A standalone file can only be generated from a single Generator.');
     }
+    if (options != BuilderOptions.empty &&
```

Comment:
I think `BuilderOptions.empty` is the default thing that will end up flowing through here if the builder is simply not configured.

Ultimately it might actually be best to just make the `BuilderOptions` argument be nullable instead of defaulting to empty. Then you can check for null here which should more accurately do what we want.

The `validatedBuildExtensionsFrom` function could also then accept a nullable BuilderOptions and bail early if it gets a null.

Reply:
> make the BuilderOptions argument be nullable instead of defaulting to empty
-> commited

> The `validatedBuildExtensionsFrom` function could also then accept a nullable BuilderOptions and bail early if it gets a null.

The thing is that the `validatedBuildExtensionsFrom` function is also called by the combiningBuilder function which already sends an `optionsMap` (and not `options`). So I can't change the argument to a nullable `BuilderOptions`. I've made it nullable though to manage cases where `options` is null.